### DISCUSSION TITLE
Full name on contributors list in repository

### DISCUSSION THREAD

**Discussion by AlfieOnGit** [community/community#22135] [New to GitHub]
<p>First of all, please bear with me as I’m just starting to use GitHub now</p>
<p>So I’m trying to upload source code into a repository however the website shows my full name (since it’s on my computer). I’m not on a full-name basis with the people I’m sending my code to so two questions; Firstly would it only be visible to me as the owner of the repository? And if not then how can I censor it? (Where it shows it <a href="https://prnt.sc/t795ic" rel="nofollow noopener">https://prnt.sc/t795ic</a>)</p>
<p>Also if I’m doing this wrong, please tell me. I’m just trying to learn how to use this site. Thanks a bunch</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The screenshot looks like the name is one GitHub got from the commit log. When you make a git commit, git includes an author name and email address in the commit. This will be visible to anyone who clones the repository, so if your repository is public it’s effectively public, even though in the web interface GitHub replaces the name with the username if the mail address belongs to a user account.</p>
<p>You can set the name and email address to use using the <code>user.name</code> and user <code>user.email</code> options, for example if you’d like to use a nickname for the current repository only (add <code>--global</code> to make it the default for all your repositories):</p>
<pre><code class="lang-bash">git config user.name "Your Nickname"
</code></pre>
<p>GitHub also offers a way to hide you mail address while keeping commits associated with your account, see the <a href="https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" rel="nofollow noopener">documentation on commit email addresses</a>.</p>
<p>However, your full name is already part of existing history of your repository. If you want to get rid of that you will have to <a href="https://help.github.com/en/github/using-git/changing-author-info" rel="nofollow noopener">rewrite the history to change author information</a>.</p>

**Comment by AlfieOnGit** [comment]
<p>This worked! Thank you</p>
<p>(Also the repository was private)</p>
